libr/anal/fcn.c

```diff
@@ -353,7 +353,8 @@ static ut64 search_reg_val(RAnal *anal, ut8 *buf, ut64 len, ut64 addr, char *reg
 		0
 	};
 	ut64 ret = UT64_MAX;
-	for (offs = 0; offs < len; offs += anal->addrbytes * oplen) {
+	const int addrbytes = anal->iob.io->addrbytes;
```

Comment:
And the null chk for io? Those apis can be used standalone we thout rcore and rio

Reply:
Added